plugins/modules/accesspoint_workflow_manager.py

```diff
@@ -0,0 +1,818 @@
+                self.module.fail_json(msg=str("\n".join(errormsg)))
+            else:
+                self.validated_config = valid_param
+                self.msg = "Successfully validated playbook config params: {0}".format(str(valid_param))
+                self.log(self.msg, "INFO")
+                self.status = "success"
+                return self
+
+        except Exception as e:
+            self.log("Invalid Param provided in playbook Yml File. {0}".format(str(e)), "ERROR")
+            self.msg = "Invalid parameters in playbook: {0}".format(str("\n".join(errormsg)))
+            self.status = "failed"
+            return self
+
+    def get_want(self, ap_config):
+        """
+        Get all site-related information from the playbook needed for creation/updation/deletion of site in Cisco Catalyst Center.
+        Parameters:
+            self (object): An instance of a class used for interacting with Cisco Catalyst Center.
+            config (dict): A dictionary containing configuration information.
+        Returns:
+            self (object): An instance of a class used for interacting with Cisco Catalyst Center.
+        Description:
+            Retrieves all site-related information from playbook that is
+            required for creating a site in Cisco Catalyst Center. It includes
+            parameters such as 'site_params' and 'site_name.' The gathered
+            information is stored in the 'want' attribute for later reference.
+        """
+        for key,value in ap_config.items():
+            if key not in ("device_fields", "ap_selected_field"):
+                self.want[key] = value
+        self.log("Desired State (want): {0}".format(str(self.want)), "INFO")
+        return self
+
+    def get_have(self, input_config):
+        """
+        Get the site details from Cisco Catalyst Center
+        Parameters:
+          - self (object): An instance of a class used for interacting with Cisco Catalyst Center.
+          - input_config (dict): A dictionary containing the configuration details.
+        Returns:
+          - self (object): An instance of a class used for interacting with  Cisco Catalyst Center.
+        Description:
+            This method queries Cisco Catalyst Center to check if a specified site
+            exists. If the site exists, it retrieves details about the current
+            site, including the site ID and other relevant information. The
+            results are stored in the 'have' attribute for later reference.
+        """
+        device_exists = False
+        current_ap_config = None
+        # check if given AP config exists, if exists store current site info
+        (device_exists, current_ap_config) = self.get_current_config(input_config)
+        self.log("Current AP config details (have): {0}".format(str(current_ap_config)), "DEBUG")
+        if device_exists:
+            self.have["mac_address"] = current_ap_config.get("mac_address")
```

Comment:
Like in get_want API used have dictionary here and at last assign it to self.have

Reply:
added the same